sources/packages/backend/libs/sims-db/src/entities/cas-invoice-batch.model.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+  PrimaryGeneratedColumn,
+} from "typeorm";
+import { ColumnNames, TableNames } from "../constant";
+import { RecordDataModel } from "./record.model";
+import { User, CASInvoiceBatchApprovalStatus, CASInvoice } from ".";
+
+/**
+ * CAS batch information to group disbursement invoices.
+ */
+@Entity({ name: TableNames.CASInvoiceBatches })
+export class CASInvoiceBatch extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+  @PrimaryGeneratedColumn()
+  id: number;
+  /**
+   * Unique batch name.
+   */
+  @Column({
+    name: "batch_name",
+  })
+  batchName: string;
+  /**
+   * Batch date.
+   */
+  @Column({
+    name: "batch_date",
+    type: "date",
+  })
+  batchDate: string;
+  /**
+   * Approval status.
+   */
+  @Column({
+    name: "approval_status",
+    type: "enum",
+    enum: CASInvoiceBatchApprovalStatus,
+    enumName: "CASInvoiceBatchApprovalStatus",
+  })
+  approvalStatus: CASInvoiceBatchApprovalStatus;
+  /**
+   * Last date and time when the status changed.
+   */
+  @Column({
+    name: "approval_status_updated_on",
+    type: "timestamptz",
+  })
+  approvalStatusUpdatedOn: Date;
+  /**
+   * User that changed the status last time.
+   */
+  @ManyToOne(() => User)
```

Comment:
I'm not sure about the default value of nullable in relational decorators. If that is not sure we can mention on relations explicitly. 

```ts
 @ManyToOne(() => User, { nullable: false })
  @JoinColumn({
    name: "approval_status_updated_by",
    referencedColumnName: ColumnNames.ID,
  })
  approvalStatusUpdatedBy: User;
```

Reply:
If the property is required. Please take care of other places as well.

Reply:
Added the `nullable: false` since the typeorm docs state otherwise as default.